### PR TITLE
Files: remove legacy code for handling media uploads

### PR DESCRIPTION
**Pull request by rinatkhaziev**
## Description

We had quite a few lines of cruft/dead code, it's time to say goodbye to it.

- This supersedes #1012 (Globally enable Photon's local filters)
- Removes VIP Legacy Filesystem implementation.

This needs quite a bit of manual testing.

### Files: remove legacy code

We've removed all legacy code related to previous iterations of media implementation.

## Checklist

Please make sure the items below have been covered before requesting a review:

- [ ] This change works and has been tested locally (or has an appropriate fallback).
- [ ] This change works and has been tested on a Go sandbox.
- [ ] This change has relevant unit tests (if applicable).
- [ ] This change has relevant documentation additions / updates (if applicable).
- [ ] I've created a changelog description that aligns with the provided examples.

## Steps to Test

1. Check out PR.
1. Test uploads locally
1. Test uploads on a sandbox
1. Verify resizing works as expected.

### CODE REVIEW

**Comment by mjangda** [comment]
Two thoughts:

- We should do a quick scan to see if any of the functions being removed are being used directly in customer code. Technically, that would be `_doing_it_wrong` since a8c-files was never meant as a public API but the code has been around since the early days and it was the wild west back then :)
- The Photon changes should be in a separate PR.

**Comment by rinatkhaziev** [comment]
> The Photon changes should be in a separate PR.

@mjangda I'm not sure I agree, both of the changes fit in the same scope of un-crufting a8c-files.php implementation. If you insist, I can do that though :)

**Comment by mjangda** [comment]
Are all sites using Jetpack Photon filters now? If not, we're introducing a functional change, and not just removing cruft.

**Comment by rinatkhaziev** [comment]
@mjangda fair point.

**Comment by rinatkhaziev** [comment]
@mjangda I reverted the Photon part, this should limit the potential blast radius

File: a8c-files.php [hunk 1]
```diff
@@ -68,12 +68,8 @@ function __construct() {
 			return 1073741824; // pow( 2, 30 )
 		});
 
-		// Conditionally load either the new Stream Wrapper implementation or old school a8c-files.
-		// The old school implementation will be phased out soon.
 		if ( defined( 'VIP_FILESYSTEM_USE_STREAM_WRAPPER' ) && true === VIP_FILESYSTEM_USE_STREAM_WRAPPER ) {
```
[the remark this answers is not in the document]
Comment (mjangda) [reply a8c-files.php:71]: I guess we need it for local compat?

File: a8c-files.php [hunk 2]
```diff
@@ -334,364 +200,6 @@ function attachment_file_exists( $file_url ) {
...
-
-	private function get_upload_path() {
-		$upload_path = trim( get_option( 'upload_path' ) );
-		if ( empty( $upload_path ) )
-			return 'wp-content/uploads';
-		else
-			return $upload_path;
-	}
-
 	function get_files_service_hostname() {
```
Comment (mjangda) [inline a8c-files.php:203]: If we remove `attachment_file_exists`, I believe this can be removed as well?

**Review by mjangda — COMMENTED** [review]
Spotted a few small things.

File: a8c-files.php [hunk 3]
```diff
@@ -199,116 +175,6 @@ private function use_jetpack_photon() {
...
-	}
-
-	function get_upload_dir( $upload ) {
-		$upload['path'] = constant( 'LOCAL_UPLOADS' ) . $upload['subdir'];
-		$upload['basedir'] = constant( 'LOCAL_UPLOADS' );
-
-		return $upload;
-	}
-
 	function attachment_file_exists( $file_url ) {
```
[the remark this answers is not in the document]
Comment (pauarge) [reply a8c-files.php:178]: 👍

**Review by pauarge — CHANGES_REQUESTED** [review]
Looks good to me except from the couple of functions that can be removed that Mo already mentioned.

File: a8c-files.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (rinatkhaziev) [reply a8c-files.php:71]: I think so?

**Review by pauarge — APPROVED** [review]
LGTM

**Comment by rinatkhaziev** [comment]
r1759-stacks

**Pull request merged**